### DISCUSSION TITLE
Use a proxy

### DISCUSSION THREAD

**Discussion by array05** [norkunas/youtube-dl-php#179] [Q&A] [answered]
Is the use of a proxy required? If so, how can they be implemented?

**Comment by norkunas** [comment, marked as answer]
Not necessarily, unless you will make many requests in a short time range then you could get blocked.
You can set the proxy via `Options::proxy(..)`, from the docs:

> To enable SOCKS proxy, specify a proper scheme. For example **socks5://user:pass@127.0.0.1:1080/**
